common/core/desktop/tests/unit/kmx/kmx_key_list.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright:    © 2018 SIL International.
+  Description:  Tests for key rules list generation in the kmx processor
+  Create Date:  30 Oct 2018
+  Authors:      Ross Cruickshank (RC) Marc Durdin (MD), Tim Eves (TSE)
+
+*/
+
+#include <kmx/kmx_processevent.h>
+
+#include "path.hpp"
+#include "state.hpp"
+
+#include "../test_assert.h"
+#include "../test_color.h"
+
+using namespace km::kbp::kmx;
+
+km_kbp_option_item test_env_opts[] =
+{
+  KM_KBP_OPTIONS_END
+};
+
+int error_args() {
+    std::cerr << "kmx: Not enough arguments." << std::endl;
+    return 1;
+}
+
+km_kbp_keyboard_key kb_key_expected_list[] = {{49,16384},{50,16384},{66,16416},{67,16384},{67,16448},{97,0},{98,0}};
```

Comment:
Should I do this test better and add a header comment that kmx.cpp test file can use, with all the other tests?
c keys: [LCTRL SHIFT K_A][SHIFT K_B][RCTRL SHIFT K_C][K_SPACE][SHIFT K_D][SHIFT K_E][SHIFT K_F]
c expected: \u0E01\u0E02\u0E03\u0020\u0E04\u0E05\u0E06
c context: 
c key_list:

Reply:
> Should I do this test better and add a header comment that kmx.cpp test file can use, with all the other tests?

I think it's probably overkill for now.

> My keyboard is totally bogus. I guess I should make a real keyboard.

The keyboard doesn't need to be sensible, just needs to exercise the functionality :grin: